source/preprocessor.tex

```diff
@@ -752,6 +833,364 @@
+
+\rSec2[cpp.embed.gen]{General}
+
+\pnum
+A preprocessing directive of the form
+\begin{ncsimplebnf}
+\terminal{\# embed <} h-char-sequence \terminal{>} \opt{pp-tokens} new-line
+\end{ncsimplebnf}
+searches a sequence of
+\impldef{sequence of places searched for an embedded resource}
+places for a resource identified uniquely by the specified sequence between
+the \tcode{<} and \tcode{>} delimiters.
+How the places are specified or the resource identified is
+\impldef{search locations for embedded resources specified with \tcode{<>}}.
+
+\pnum
+A preprocessing directive of the form
+\begin{ncsimplebnf}
+\terminal{\# embed "} q-char-sequence \terminal{"} \opt{pp-tokens} new-line
+\end{ncsimplebnf}
+searches for a resource identified by the specified sequence between the
+\tcode{"} delimiters.
+The named resource is searched for in an
+\impldef{manner of search for named resource}
+manner.
+If this search is not supported, or if the search fails, the directive is
+reprocessed as if it read
+\begin{ncsimplebnf}
+\terminal{\# embed <} h-char-sequence \terminal{>} \opt{pp-tokens} new-line
+\end{ncsimplebnf}
+with the identical contained sequence (including \tcode{>} characters, if any)
+from the original directive.
+
+\pnum
+\recommended A mechanism similar to, but distinct from, the
+\impldef{sequence of places searched for a header}
+search paths used for \tcode{\#include} \iref{cpp.include}
+is encouraged.
+
+\pnum
+Either form of the \tcode{\#embed} directive processes the
+\grammarterm{pp-tokens}, if present, just as in normal text.
+The \grammarterm{pp-tokens} shall then have the form
+\grammarterm{embed-parameter-seq}.
+
+\pnum
+A resource is a source of data accessible from the translation environment.
+A resource has an \gterm{implementation-resource-width}, which is the
+\impldef{size in bits of a resource}
+size in bits of the resource.
+If the \gterm{implementation-resource-width} is not an integral multiple of
+\libmacro{CHAR_BIT}, the program is ill-formed.
+Let \gterm{implementation-resource-count} be
+\gterm{implementation-resource-width} divided by \libmacro{CHAR_BIT}.
+Every resource also has a \gterm{resource-count}, which is
+
+\begin{itemize}
+\item
+  the value as computed from the optionally-provided \tcode{limit}
+  \grammarterm{embed-parameter}\iref{cpp.embed.param.limit}, if present;
+\item
+  otherwise, the \gterm{implementation-resource-count}.
+\end{itemize}
+
+A resource is empty if the resource-count is zero.
+
+\pnum
+\begin{example}
+\begin{codeblock}
+// ill-formed if the implementation-resource-width is 6 bits
+#embed "6_bits.bin"
+\end{codeblock}
+\end{example}
+
+\pnum
+The \tcode{\#embed} directive is replaced by a comma-delimited list of integer
+literals of type \tcode{int}, unless otherwise modified by embed
+parameters\iref{cpp.embed.param}.
+
+\pnum
+The integer literals in the comma-delimited list correspond to
+\gterm{resource-count} consecutive calls to \tcode{std::fgetc} \iref{cstdio.syn}
```

Comment:
Same here: Switch to \defn; no italics here (and there should be none in the paper, but maybe we missed an occurrence during CWG review).